src/lava/proc/io/bridge/utils.py

```diff
@@ -0,0 +1,275 @@
+                      ChannelRecvBufferNotEmpty):
+        raise TypeError(
+            "Expected <channel_config>.recv_buffer_not_empty "
+            "to be of type ChannelRecvBufferNotEmpty. Got "
+            "<channel_config>.recv_buffer_not_empty = "
+            f"{channel_config.recv_buffer_not_empty}.")
+
+
+def validate_send_data(data: np.ndarray, out_port: OutPort) -> None:
+    """Validate that the data is of the same shape as the OutPort.
+
+    Parameters
+    ----------
+    data : np.ndarray
+        Data to be sent.
+    out_port : OutPort
+        OutPort through which the data will ultimately be sent.
+    """
+    if not isinstance(data, np.ndarray):
+        raise TypeError("Expected <data> to be of type np.ndarray. Got "
+                        f"<data> = {data}")
+
+    if data.shape != out_port.shape:
+        raise ValueError("Expected <data>.shape to be equal to shape of "
+                         f"OutPort. Got <data>.shape = {data.shape} and "
+                         f"<out_port>.shape = {out_port.shape}.")
+
+
+def send_data_blocking(src_port: CspSendPort, data: np.ndarray) -> None:
+    """Send data through the src_port, when using
+    ChannelSendBufferFull.BLOCKING.
+
+    If the channel buffer is full, this method blocks.
+
+    Parameters
+    ----------
+    src_port : CspSendPort
+        Port through which the data is sent.
+    data : np.ndarray
+        Data to be sent.
+    """
+    src_port.send(data)
+
+
+def send_data_non_blocking_drop(src_port: CspSendPort, data: np.ndarray) -> \
+        None:
+    """Send data through the src_port, when using
+    ChannelSendBufferFull.NON_BLOCKING_DROP.
+
+    If the channel buffer is full, the data is not sent, a warning is issued,
+    and this method does not block.
+
+    Parameters
+    ----------
+    src_port : CspSendPort
+        Port through which the data is sent.
+    data : np.ndarray
+        Data to be sent.
+    """
+    if src_port.probe():
+        src_port.send(data)
+    else:
+        warnings.warn("Send buffer is full. Dropping items ...")
+
+
+def recv_empty_blocking(dst_port: CspRecvPort, **kwargs) -> np.ndarray:
+    """Receive data through the dst_port, when using
+    ChannelRecvBufferEmpty.BLOCKING.
+
+    This method is blocking.
+
+    Parameters
+    ----------
+    dst_port : CspRecvPort
+        Port through which the data is received.
+
+    Returns
+    ----------
+    data : np.ndarray
+        Data received.
+    """
+    return dst_port.recv()
+
+
+def recv_empty_non_blocking_zeros(zeros: np.ndarray, **kwargs) \
+        -> np.ndarray:
+    """Receive data through the dst_port, when using
+    ChannelRecvBufferEmpty.NON_BLOCKING_ZEROS.
+
+    This method returns zeros.
+
+    Parameters
+    ----------
+    zeros : np.ndarray
+        Zeros array to be returned.
+
+    Returns
+    ----------
+    data : np.ndarray
+        Data received (zeros).
+    """
+    return zeros
```

Comment:
Is this not implemented, or is this a dummy method to satisfy a consistent interface to other recv methods?

Reply:
This is a dummy method to satisfy a consistent interface to other recv methods.